nwb_conversion_tools/datainterfaces/ecephys/axona/axonadatainterface.py

```diff
@@ -296,3 +420,205 @@ def run_conversion(self, nwbfile: NWBFile, metadata: dict):
+
+
+def get_all_filenames(filename: PathType):
+    """
+    Read LFP filenames of `.eeg` or `.egf` files in filename's directory.
+    E.g. if filename='/my/directory/my_file.eeg', all .eeg channels will be
+    appended to the output.
+
+    Parameters:
+    -----------
+    filename : path-like
+        Full filename of either .egg or .egf file
+
+    Returns:
+    --------
+    path_list : list
+        List of filenames
+    """
+
+    suffix = Path(filename).suffix[0:4]
+    current_path = Path(filename).parent
+
+    path_list = [cur_path.name for cur_path in Path(filename).parent.rglob("*" + suffix + "*")]
+
+    return path_list
+
+
+def read_all_eeg_file_lfp_data(filename: PathType):
+    """
+    Read LFP data from all Axona `.eeg` or `.egf` files in filename's directory.
+    E.g. if filename='/my/directory/my_file.eeg', all .eeg channels will be conactenated
+    to a single np.array (chans x nobs). For .egf files substitude the file suffix.
+
+    Parameters:
+    -------
+    filename (Path or Str):
+        Full filename of Axona `.eeg` or `.egf` file.
+
+    Returns:
+    -------
+    np.array (chans x obs)
+    """
+
+    filename_list = get_all_filenames(filename)
+    parent_path = Path(filename).parent
+
+    eeg_memmaps = list()
+    sampling_rates = set()
+    for fname in filename_list:
+
+        sampling_rates.add(get_eeg_sampling_frequency(parent_path / fname))
+
+        eeg_memmaps.append(read_eeg_file_lfp_data(parent_path / fname))
+
+    assert len(sampling_rates) < 2, "File headers specify different sampling rates. Cannot combine EEG data."
+
+    eeg_data = np.concatenate(eeg_memmaps, axis=0)
+
+    return eeg_data
+
+
+class AxonaLFPDataInterface(AxonaRecordingExtractorInterface):
+    """..."""
+
+    @classmethod
+    def get_source_schema(cls):
+        return dict(
+            required=["filename"],
+            properties=dict(filename=dict(type="string")),
+            type="object",
+            additionalProperties=False,
+        )
+
+    def __init__(self, filename: PathType):
+        self.recording_extractor = se.NumpyRecordingExtractor(
+            timeseries=read_all_eeg_file_lfp_data(filename),
+            sampling_frequency=get_eeg_sampling_frequency(filename),
+        )
+        self.subset_channels = None
+        self.source_data = dict(filename=filename)
+
+    def get_metadata_schema(self):
+        metadata_schema = super().get_metadata_schema()
+        metadata_schema["properties"]["Ecephys"]["properties"].update(
+            ElectricalSeries_lfp=get_schema_from_hdmf_class(ElectricalSeries)
+        )
+        return metadata_schema
+
+    def get_metadata(self):
+        """Retrieve Ecephys metadata specific to the Axona format."""
+        metadata = super().get_metadata()
+        metadata["Ecephys"].pop("ElectricalSeries_raw", None)
+        metadata["Ecephys"].update(ElectricalSeries_lfp=dict(name="LFP", description="Local field potential signal."))
+
+        return metadata
```

Comment:
```suggestion
```
based on the rationale of the comment above, this would be unnecessary too.
Just like you called the `run_conversion()` below with `write_as='lfp'` is sufficient

Reply:
This was a bit tricker. Originally, I had inherited from `AxonaRecordingExtractorInterface`, but it is more intuitive to inherit from `BaseLFPExtractorInterface`. However, `BaseLFPExtractorInterface` did not update the metadata_schema to include `ElectricalSeries_lfp`, yielding an error. I have now updated `BaseLFPExtractorInterface` accordingly [here](https://github.com/catalystneuro/nwb-conversion-tools/pull/233) and this simplifies the code as you suggest.